value/src/main/java/com/google/auto/value/extension/AutoValueExtension.java

```diff
@@ -108,6 +108,38 @@
     Set<ExecutableElement> abstractMethods();
   }
 
+  /**
+   * Indicates to an annotation processor environment supporting incremental annotation processing
+   * (currently a feature specific to Gradle starting with version 4.8) the incremental type of an
+   * Extension.
+   *
+   * <p>The constants for this enum are ordered by increasing performance (but also constraints).
+   *
+   * @see <a
+   *     href="https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_annotation_processing">Gradle
+   *     documentation of its incremental annotation processing</a>
+   */
+  public enum IncrementalExtensionType {
+    UNKNOWN,
```

Comment:
I feel like the individual enum constants should have an understandable summary of what they mean. Here's what I've written in the copy of this PR that I'm importing into Google's private repo:

```java
    /**
     * The incrementality of this extension is unknown, or it is neither aggregating nor isolating.
     */
    UNKNOWN,

    /**
     * This extension is <i>aggregating</i>, meaning that it may generate outputs based on several
     * annotated input classes and it respects the constraints imposed on aggregating processors.
     * It is unusual for AutoValue extensions to be aggregating.
     *
     * @see <
     *     href="https://docs.gradle.org/current/userguide/java_plugin.html#aggregating_annotation_processors">Gradle
     *     definition of aggregating processors</a>
     */
    AGGREGATING,

    /**
     * This extension is <i>isolating</i>, meaning roughly that its output depends on the
     * {@code @AutoValue} class and its dependencies, but not on other {@code @AutoValue} classes
     * that might be compiled at the same time. The constraints that an isolating extension must
     * respect are the same as those that Gradle imposes on an isolating annotation processor.
     *
     * @see <
     *     href="https://docs.gradle.org/current/userguide/java_plugin.html#isolating_annotation_processors">Gradle
     *     definition of aggregating processors</a>
     */
    ISOLATING
```

Does that seem right? I'm actually finding it hard to imagine what an aggregating extension might be trying to do.

Reply:
> Does that seem right?

SGreatTM :wink:

> I'm actually finding it hard to imagine what an aggregating extension might be trying to do.

IIUC, looking up another class by name, e.g using a naming rule and/or an annotation property value (or processor option).
Probably unusual, but not impossible… :man_shrugging: